text/0000-Project-Goals-2024h2.md

```diff
@@ -0,0 +1,521 @@
+
+[RFL.com]: https://rust-for-linux.com/
+[RFL#2]: https://github.com/Rust-for-Linux/linux/issues/2
+
+
+### Project goals
+
+The slate of additional project goals are as follows. These goals all have identified owners who will drive the work forward as well as a viable work plan. The goals include asks from the listed Rust teams, which are cataloged in the [reference-level explanation](#reference-level-explanation) section below. Some goals are actively looking for volunteers; these goals are tagged with ![Help wanted][].
+
+| Goal                                                                                                      | Owner                | Team                                                         |
+| ---                                                                                                       | ---                  | ---                                                          |
+| [Const traits](https://rust-lang.github.io/rust-project-goals/2024h2/const-traits.html)                                                                           | [Deadbeef][]           | [lang], [types]                                              |
+| [Patterns of empty types](https://rust-lang.github.io/rust-project-goals/2024h2/Patterns-of-empty-types.html)                                                     | [@Nadrieril][]           | [compiler], [lang]                                           |
+| [Stabilize cargo-script](https://rust-lang.github.io/rust-project-goals/2024h2/cargo-script.html)                                                                 | [Ed Page][]               | [cargo], [lang]                                              |
+| [Next-generation trait solver](https://rust-lang.github.io/rust-project-goals/2024h2/next-solver.html)                                                            | [lcnr][]                | [rust-analyzer], [types]                                     |
+| [Assemble project goal slate](https://rust-lang.github.io/rust-project-goals/2024h2/Project-goal-slate.html)                                                      | [Niko Matsakis][]        | [leadership-council]                                         |
+| [Optimizing Clippy & linting](https://rust-lang.github.io/rust-project-goals/2024h2/optimize-clippy.html)                                                         | [Alejandra González][]             | [clippy]                                                     |
+| [Make Rustdoc Search easier to learn](https://rust-lang.github.io/rust-project-goals/2024h2/rustdoc-search.html)                                                  | [Michael Howell][]           | [infra], [leadership-council], [rustdoc], [rustdoc-frontend] |
```

Comment:
It's not clear to me why this goal needs the infra-team. Reading through the proposal, it seems that the rustdoc and/or docs.rs teams are in a much better position to provide feedback.

Reply:
cc @GuillaumeGomez

Reply:
oops, sorry, this is @notriddle

Reply:
I guess the goal says "Collecting popular queries for review" -- I'll redirect to @rust-lang/docs-rs, I guess? I don't know how we would gather popular queries from rustdoc otherwise, unless you plan to have it "phone home".

Reply:
(You could of course use the ["telemetry without telemetry"](https://estebank.github.io/rustc-metrics.html) scheme described by @estebank, but I don't think that's what you were proposing.)

Reply:
Alternatively I could just remove this ask and we can figure it out as we go.

Reply:
Ah, I see! Thanks for the context!

For `docs.rs`, the docs.rs team should be the best point of contact. But `doc.rust-lang.org` is served as a static website from S3 and we're not collecting any logs today. So that might actually require the infra-team to help out after all.

Reply:
I somehow missed the mention above, sorry about the delay in responding. 

Docs.rs doesn't change rustdoc search in any way, so it's still only in the frontend. The only thing in our webserver (or CDN) logs would be fetching the search index file, not the search term itself. 

The only time when you would see the term in the access logs is probably when someone searches a term, and then manually refreshes the page, which feels like much less than 80% gains. 

So the only way to get some statistics on terms would be to add frontend JS code that hooks into the search itself on the frontend side, which probably would trigger the mentioned consent popup above?

Reply:
My proposal is that we remove this part of the project goal but that @notriddle you continue discussing with @syphar or @jdno to see if there's some easy step we can take here.

Reply:
Resolving since https://github.com/rust-lang/rust-project-goals/pull/51 removes this point, thanks @notriddle!